--- yosys/Makefile	2022-05-30 17:07:16.025227248 +0200
+++ yosys/Makefile	2022-05-30 17:09:07.093207560 +0200
@@ -265,18 +265,19 @@
 EMCC_LDFLAGS := --memory-init-file 0 --embed-file share
 EMCC_LDFLAGS += -s NO_EXIT_RUNTIME=1
 EMCC_LDFLAGS += -s EXPORTED_FUNCTIONS="['_main','_run','_prompt','_errmsg','_memset']"
-EMCC_LDFLAGS += -s TOTAL_MEMORY=134217728
-EMCC_LDFLAGS += -s EXPORTED_RUNTIME_METHODS='["ccall", "cwrap"]'
+EMCC_LDFLAGS += -s TOTAL_MEMORY=2621440008
+EMCC_LDFLAGS += -s EXPORTED_RUNTIME_METHODS='["callMain", "ccall", "cwrap", "FS", "IDBFS", "NODEFS", "PROXYFS", "WORKERFS"]'
+EMCC_LDFLAGS += -s MODULARIZE=1 -s EXPORT_NAME='Yosys'
 # https://github.com/kripken/emscripten/blob/master/src/settings.js
 CXXFLAGS += $(EMCC_CXXFLAGS)
 LDFLAGS += $(EMCC_LDFLAGS)
-LDLIBS =
++LDLIBS := -lproxyfs.js $(EMCC_LDLIBS)
 EXE = .js

 DISABLE_SPAWN := 1

 TARGETS := $(filter-out $(PROGRAM_PREFIX)yosys-config,$(TARGETS))
-EXTRA_TARGETS += yosysjs-$(YOSYS_VER).zip
+# EXTRA_TARGETS += yosysjs-$(YOSYS_VER).zip

 ifeq ($(ENABLE_ABC),1)
 LINK_ABC := 1
@@ -441,7 +442,7 @@

 ifeq ($(ENABLE_ZLIB),1)
 CXXFLAGS += -DYOSYS_ENABLE_ZLIB
-LDLIBS += -lz
+# LDLIBS += -lz
 endif


@@ -1010,10 +1011,10 @@
 config-emcc: clean
 	echo 'CONFIG := emcc' > Makefile.conf
 	echo 'ENABLE_TCL := 0' >> Makefile.conf
-	echo 'ENABLE_ABC := 0' >> Makefile.conf
+	echo 'ENABLE_ABC := 1' >> Makefile.conf
 	echo 'ENABLE_PLUGINS := 0' >> Makefile.conf
 	echo 'ENABLE_READLINE := 0' >> Makefile.conf
-	echo 'ENABLE_ZLIB := 0' >> Makefile.conf
+	echo 'ENABLE_ZLIB := 1' >> Makefile.conf

 config-wasi: clean
 	echo 'CONFIG := wasi' > Makefile.conf
--- yosys/kernel/register.cc	2022-03-19 18:20:08.592033745 +0100
+++ yosys/kernel/register.cc	2022-03-19 18:21:39.912096058 +0100
@@ -26,7 +26,7 @@
 #include <errno.h>

 #ifdef YOSYS_ENABLE_ZLIB
-#include <zlib.h>
+#include "libs/zlib/zlib.h"

 PRIVATE_NAMESPACE_BEGIN
 #define GZ_BUFFER_SIZE 8192
--- yosys/libs/fst/fstapi.h	2022-03-19 18:20:14.696037901 +0100
+++ yosys/libs/fst/fstapi.h	2022-03-19 18:21:56.072107115 +0100
@@ -55,7 +55,7 @@

 #include "stdint.h"
 #else
-#include <zlib.h>
+#include "libs/zlib/zlib.h"
 #include <unistd.h>
 #endif
 #include <time.h>
