--- yosys/Makefile	2022-05-30 17:07:16.025227248 +0200
+++ yosys/Makefile	2022-05-30 17:09:07.093207560 +0200
@@ -265,18 +265,19 @@
 EMCC_LDFLAGS := --memory-init-file 0 --embed-file share
 EMCC_LDFLAGS += -s NO_EXIT_RUNTIME=1
 EMCC_LDFLAGS += -s EXPORTED_FUNCTIONS="['_main','_run','_prompt','_errmsg','_memset']"
-EMCC_LDFLAGS += -s TOTAL_MEMORY=134217728
-EMCC_LDFLAGS += -s EXPORTED_RUNTIME_METHODS='["ccall", "cwrap"]'
+EMCC_LDFLAGS += -s ALLOW_MEMORY_GROWTH=1
+EMCC_LDFLAGS += -s EXPORTED_RUNTIME_METHODS='["callMain", "ccall", "cwrap", "FS", "IDBFS", "NODEFS", "PROXYFS", "WORKERFS"]'
+EMCC_LDFLAGS += -s MODULARIZE=1 -s EXPORT_NAME='Yosys' -s LLD_REPORT_UNDEFINED=1
 # https://github.com/kripken/emscripten/blob/master/src/settings.js
 CXXFLAGS += $(EMCC_CXXFLAGS)
 LDFLAGS += $(EMCC_LDFLAGS)
-LDLIBS =
++LDLIBS := -lproxyfs.js $(EMCC_LDLIBS)
 EXE = .js

 DISABLE_SPAWN := 1

 TARGETS := $(filter-out $(PROGRAM_PREFIX)yosys-config,$(TARGETS))
-EXTRA_TARGETS += yosysjs-$(YOSYS_VER).zip
+# EXTRA_TARGETS += yosysjs-$(YOSYS_VER).zip

 ifeq ($(ENABLE_ABC),1)
 LINK_ABC := 1
@@ -1010,10 +1011,10 @@
 config-emcc: clean
 	echo 'CONFIG := emcc' > Makefile.conf
 	echo 'ENABLE_TCL := 0' >> Makefile.conf
-	echo 'ENABLE_ABC := 0' >> Makefile.conf
+	echo 'ENABLE_ABC := 1' >> Makefile.conf
 	echo 'ENABLE_PLUGINS := 0' >> Makefile.conf
 	echo 'ENABLE_READLINE := 0' >> Makefile.conf
	echo 'ENABLE_ZLIB := 0' >> Makefile.conf

 config-wasi: clean
 	echo 'CONFIG := wasi' > Makefile.conf
